build failure on arm

Using fresh clone of master at db11bc7.  My last successful build was at e18ed61.
```
/home/sachs/src/julia-master/base/precompile.jl

signal (11): Segmentation fault
while loading no file, in expression starting on line 0
Allocations: 44889342 (Pool: 44888794; Big: 548); GC: 25
Segmentation fault
*** This error is usually fixed by running `make clean`. If the error persists, try `make cleanall`. ***
Makefile:229: recipe for target '/home/sachs/src/julia-master/usr/lib/julia/sys.o' failed
make[1]: *** [/home/sachs/src/julia-master/usr/lib/julia/sys.o] Error 1
Makefile:99: recipe for target 'julia-sysimg-release' failed
make: *** [julia-sysimg-release] Error 2
```

On arm, with release-0.5, I still get a failure:
```

signal (11): Segmentation fault
while loading errno_h.jl, in expression starting on line 1
Allocations: 5705285 (Pool: 5705159; Big: 126); GC: 15
Segmentation fault
*** This error is usually fixed by running `make clean`. If the error persists, try `make cleanall`. ***
make[1]: *** [/home/viral/julia/usr/lib/julia/sys.o] Error 1
make: *** [julia-sysimg-release] Error 2
```

--- a/src/cgmemmgr.cpp
+++ b/src/cgmemmgr.cpp
@@ -462,11 +462,12 @@ public:
     virtual ~ROAllocator() {}
     virtual void finalize()
     {
-        if (exec) {
-            for (auto &alloc: allocations) {
-                sys::Memory::InvalidateInstructionCache(alloc.rt_addr,
-                                                        alloc.sz);
-            }
+        for (auto &alloc: allocations) {
+            // ensure the mapped pages are consistent
+            sys::Memory::InvalidateInstructionCache(alloc.wr_addr,
+                                                    alloc.sz);
+            sys::Memory::InvalidateInstructionCache(alloc.rt_addr,
+                                                    alloc.sz);
         }
         completed.clear();
         allocations.clear();

